### PR TITLE
Type inference endpoint

### LINKED ISSUE #200: Table creation from data file should infer data types automatically
**Problem**

Currently, we create an all-string table when we create a table from a data file. In order to support our [design for file import](https://wiki.mathesar.org/design/process/documents/table-import-specs), we'll need to show the user the types we recommend for each column.

**Proposed solution**

We should have an API that provides inferred data types for each column in a table. We don't actually want to change the table yet, just infer data types for each column.

**Additional context**

See #92 for DB functions to be used for this.

Related to #199, although I don't think it's actually blocked by it.

### PR DESCRIPTION
**Pull request by eito-fis**
Fixes #200 

Adds an endpoint (`/api/v0/tables/{table.id}/type_suggestions/`) which returns a dictionary mapping column names to inferred types.

**Technical details**
The endpoint calls the `infer_table_column_types()` function from #268, then converts the SQLAlchemy types into strings and returns the values. 

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [ ] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by eito-fis** [comment]
Currently blocked by #283, should be resolved by #284

File: mathesar/tests/views/api/test_table_api.py
Comment (kgodey) [inline mathesar/tests/views/api/test_table_api.py:129]: The way types are represented here should match the way they're represented in the current table list/detail API. We use "INTEGER", "VARCHAR", etc.

File: mathesar/tests/views/api/test_table_api.py
Comment (kgodey) [inline mathesar/tests/views/api/test_table_api.py:124]: I assume you'll add some error case tests as well.

File: mathesar/tests/views/api/test_table_api.py
Comment (kgodey) [inline mathesar/tests/views/api/test_table_api.py:137]: We shouldn't be using the legacy function in new code since it's scheduled to be deleted.

File: mathesar/tests/views/api/test_table_api.py
Comment (kgodey) [inline mathesar/tests/views/api/test_table_api.py:147]: I assume this is for debugging will be removed in the final PR.

**Review by kgodey — COMMENTED** [review]
I added a couple of comments. I know this PR isn't ready for review yet, but hopefully they're helpful for the final version.

File: mathesar/tests/views/api/test_table_api.py
Comment (eito-fis) [reply mathesar/tests/views/api/test_table_api.py:137]: Do we just want to use the table creation API instead?

File: mathesar/tests/views/api/test_table_api.py
Comment (eito-fis) [reply mathesar/tests/views/api/test_table_api.py:124]: I added a test for an invalid table - is there anything else to look for?

**Comment by eito-fis** [comment]
Updated to address comments, tests should be passing once we get #284 merged.

File: mathesar/tests/views/api/test_table_api.py
Comment (kgodey) [reply mathesar/tests/views/api/test_table_api.py]: Yes.

**Comment by eito-fis** [comment]
Blocked waiting for #268 to me merged, but is ready for review.

File: mathesar/tests/views/api/test_table_api.py
Comment (kgodey) [reply mathesar/tests/views/api/test_table_api.py]: Are there instances where the type suggestions would fail? That would be worth testing.

I'll merge the PR, if you think of any new tests, make a new PR. :)

**Review by kgodey — APPROVED** [review]

**Pull request merged**
